samples/bluetooth/central_hr/CMakeLists.txt

```diff
@@ -1,6 +1,9 @@
 # SPDX-License-Identifier: Apache-2.0
 
 cmake_minimum_required(VERSION 3.20.0)
+
+set(ENV{FILE_SUFFIX} "${FILE_SUFFIX}")
```

Comment:
this is weird.. and I'm not really sure why you are doing this instead of just using another kconfig option, or different kconfig files

Reply:
Should not be setting environmental values when you build a sample, that is polluting the workspace

Reply:
> this is weird.. and I'm not really sure why you are doing this instead of just using another kconfig option, or different kconfig files

I'm not sure how a solution like this would look like. How would you do it?

Reply:
> Should not be setting environmental values when you build a sample, that is polluting the workspace

Environment variables are only passed on to child processes. It will not pollute the shell variables (I assume that's what 'workspace' means).